### DISCUSSION TITLE
Can I send a new task to be executed within a task without waiting to finish it?

### DISCUSSION THREAD

**Discussion by Scolymus** [celery/celery#7682] [Q&A] [answered]
Suppose we have Task A using FastAPI+celery:
```
@app.post("/send_A")
def send_A(myvalue: int = None):
    task = ask_A.delay(myvalue)
    return {"task_id": task.id}

@celery.task(name="ask_A", bind=True)
def ask_A(self, myvalue):
    # DO STUFF
    # NEW TASK
    return {'STATE': 'DONE'}

@app.post("/get_result_A")
def get_result_A(task_id):
    task_result = celery.AsyncResult(task_id)
```

Now, I want to execute a new task in a celery worker that will run in a different queue from the one used for task A, where it says `# NEW TASK`. However, task A should finish without waiting for the result of this new task so a new task A can be taken from the queue for A tasks. Is it possible to do this?

The idea is that process A will do some task, and then at the end of A I will execute a new function to download a big file that must be queued, but I don't want to wait to download this file, and tasks A and downloads should have two different queues.

**Comment by olii** [comment, marked as answer]
Yes, just add a new task and call it in the body of `task A`:
```python
@celery.task(name="ask_A", bind=True)
def ask_A(self, myvalue):
    ...
    some_other_task.delay(...)
    return {'STATE': 'DONE'}

@celery.task
def some_other_task(...):
    ...
```

To separate tasks into different queues, you need to set up [task routing](https://docs.celeryq.dev/en/stable/userguide/configuration.html#std-setting-task_routes).
